pyvista/core/composite.py

```diff
@@ -690,3 +703,238 @@ def copy(self, deep=True):
         newobject.copy_meta_from(self, deep)
         newobject.wrap_nested()
         return newobject
+
+    def set_active_scalars(
+        self, name: Optional[str], preference: str = 'cell', allow_missing: bool = False
+    ) -> Tuple[FieldAssociation, np.ndarray]:  # type: ignore
+        """Find the scalars by name and appropriately set it as active.
+
+        To deactivate any active scalars, pass ``None`` as the ``name``.
+
+        Parameters
+        ----------
+        name : str or None
+            Name of the scalars array to assign as active.  If
+            ``None``, deactivates active scalars for both point and
+            cell data.
+
+        preference : str, optional
+            If there are two arrays of the same name associated with
+            points or cells, it will prioritize an array matching this
+            type.  Can be either ``'cell'`` or ``'point'``.
+
+        allow_missing : bool, optional
+            Allow missing scalars in part of the composite dataset. If all
+            blocks are missing the array, it will raise a ``KeyError``.
+
+        Returns
+        -------
+        pyvista.FieldAssociation
+            Field association of the scalars activated.
+
+        numpy.ndarray
+            An array from the dataset matching ``name``.
+
+        Notes
+        -----
+        The number of components of the data must match.
+
+        """
+        data_assoc: List[Tuple[FieldAssociation, np.ndarray, MultiBlock]] = []
+        for block in self:
+            if block is not None:
+                if isinstance(block, MultiBlock):
+                    field, scalars = block.set_active_scalars(name, preference, allow_missing=True)
+                else:
+                    try:
+                        field, scalars = block.set_active_scalars(name, preference)
+                    except KeyError:
+                        if not allow_missing:
+                            raise
+                        block.set_active_scalars(None, preference)
+                        field, scalars = FieldAssociation.NONE, np.array([])
+
+                if field != FieldAssociation.NONE:
+                    data_assoc.append((field, scalars, block))
+
+        if name is None:
+            return FieldAssociation.NONE, np.array([])
+
+        if not data_assoc:
+            raise KeyError(f'"{name}" is missing from all the blocks of this composite dataset.')
+
+        # Verify that all the data has identical field association
+        first_field = data_assoc[0][0]
```

Comment:
Why is this section of testing necessary for setting active scalars?  It seems we'd need to do this for a particular use of the active scalars, but it isn't required to set them.

Reply:
Agreed. I've refactored most of this out in 416ef55af169ce2da8d4443e3c1d17eeff9d7364, but it's still important to know the field association made active when setting the scalars in order to set it in the mapper, so I've kept in a tiny bit of the logic.